docs/developer/architecture/configurations.rst

```diff
@@ -91,17 +91,45 @@ requirements for the following reasons:
 How it works
 ^^^^^^^^^^^^
 
-During the Bootstrap stages, when we are assertive that the K8s cluster is
-fully ready and available we could perform the following actions:
+During Bootstrap, Upgrade or Downgrade stages, when we are assertive that
+the K8s cluster is fully ready and available we could perform the following
+actions:
 
-  - Create and deploy ConfigMaps that hold cluster and service configurations
-    and pre-fill them with default values.
-  - Template service pods and deployments to consume configuration data
-    directly from the above deployed ConfigMaps
+  - Firstly, create and deploy ConfigMaps that will hold customizable cluster
+    and service configurations.
+    These ConfigMaps should define an empty `config.yaml` in the data section
+    of the ConfigMap for later use. A standard layout for each customizable
+    field could be added in the documentation to assist MetalK8s administrator
+    in adding and modifying customizations.
+
+  - In a `csc.yaml` file located at the root directory of each MetalK8s Addon,
+    define the keys and values for default service configurations in a
+    YAML structured format.
+
+      - The layout of service configurations within this file could have root
+        key being the impacted ConfigMap name. For example:
+
+    .. code-block:: yaml
+
+        metalk8s-prometheus-config:
+          spec:
+            deployment:
+              replicas: 1
+
+  - During Addon manifests rendering, call a Salt module that will merge
+    the configurations defined within the customizable ConfigMap to those
+    defined in `csc.yaml` using the recommended merge strategy.
+
+    The resulting configuration (a python dict object) will be used to populate
+    the desired configuration fields within each Addon chart at render time.
 
 This approach works because in a MetalK8s cluster, ConfigMaps for cluster and
 service configurations are available before we deploy the configured services.
```

Comment:
This is not really a relevant comment anymore, right? Also, what about the behavior if the ConfigMaps are missing? Can we fallback on defaults only, or should we fail?

Reply:
Yes, we will fall back to defaults if CSC Configmaps are empty. But if Configmaps are missing because an Admin deleted them, then the CSC module will fail then raise an error